epochX/cudacpp/CODEGEN/PLUGIN/CUDACPP_SA_OUTPUT/madgraph/iolibs/template_files/gpu/process_matrix.inc

```diff
@@ -7,6 +7,32 @@
 ! Further modified by: J. Teig, A. Valassi (2021-2023) for the MG5aMC CUDACPP plugin.
 !==========================================================================
 
+#if not defined MGONGPUCPP_GPUIMPL and defined MGONGPU_FPTYPE2_FLOAT
+      // Avoid underflow FPE #831 for jamp (flush-to-zero jamp values whose square is below FLT_MIN)
+      constexpr fptype2 minjamp = 1.1 * constexpr_sqrt( FLT_MIN );
```

Comment:
> if underflow occurs, it is not an issue (and even good) if we keep some printout to warn the user.

This is done. I have added printouts if underflow occurs in cudacpp.

> Could we change this in two ways:
> 
>     1. use a relative threshold compare to the maximum of the real/imaginary value of the jamp_sv
> 
>     2. put the threshold related to the numerical error of double precision (so around 1e-15)
> 
> 
> And we can made the code safer by
> 
>     1. one can rescale all jamp_sv by dividing by the max
> 
>     2. remultiply the final ztemp by the max square

Conversely, I have completely removed the code, for various reasons
- One, the new general strategy, after discussing with you, is do not try to prevent underflows, just let them happen and report them
- Two, after some tests, I realised that even FLT_MIN/10 is actually printed out to a reasonable value. I think the underflow just produces a 'denormal' with less precision than a float, but even usable, for some underflows. I prefer to just keep the code simpler at this stage.

Resolving